hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDiscoveryServiceBuilder.java

```diff
@@ -162,13 +159,19 @@ private AddressProvider createAddressProvider(ClientConfig clientConfig, Discove
     }
 
     private boolean discoverySpiEnabled(ClientNetworkConfig networkConfig) {
-        return (networkConfig.getDiscoveryConfig() != null && networkConfig.getDiscoveryConfig().isEnabled())
-                || Boolean.parseBoolean(properties.getString(DISCOVERY_SPI_ENABLED));
+        DiscoveryConfig discoveryConfig = networkConfig.getDiscoveryConfig();
+        if (discoveryConfig.getDiscoveryServiceProvider() != null) {
+            return true;
+        }
+        Collection<DiscoveryStrategyConfig> discoveryStrategyConfigs = discoveryConfig.getDiscoveryStrategyConfigs();
+        if (discoveryStrategyConfigs.stream().anyMatch(DiscoveryStrategyConfig::isEnabled)) {
+            return true;
+        }
+        return networkConfig.getDiscoveryConfig() != null && networkConfig.getDiscoveryConfig().isEnabled();
     }
 
     private boolean usePublicAddress(ClientConfig config) {
-        return properties.getBoolean(ClientProperty.DISCOVERY_SPI_PUBLIC_IP_ENABLED)
```

Comment:
If we remove `DISCOVERY_SPI_PUBLIC_IP_ENABLED`, then how do you enable client public discovery for custom discovery strategies (not aliased, but for example, written by the community)?

Reply:
Then, We either will not remove `DISCOVERY_SPI_PUBLIC_IP_ENABLED` or 
We will introduce `use-public-api` for `DiscoveryStrategyConfig`. 

When I saw the related javadoc I thought that these properties are not used. But it seems they have usages. 

Similarly I have introduced setEnabled `DiscoveryStrategyConfig` just because I have removed the property. 
So, this is not a cleanup anymore, we are replacing the property with config. In that case, may be we should not delete these properties but just update the javadoc . @leszko Wdyt ?

Reply:
yeah, agree, I would not remove them for now.